### PR TITLE
[MI-2419][server]: Added support to approve/reject run stage pipeline events.

### PR DESCRIPTION
**Pull request by avas27JTG**
Added support to approve/reject run stage pipeline events with the following approver's scenarios:

- nth number in any order
- all in any order
- all in sequence
- anyone's rejection

### CODE REVIEW

File: server/plugin/api.go
Comment (ayusht2810) [inline server/plugin/api.go:823]: are we sure about the length here?

File: server/plugin/api.go
Comment (ayusht2810) [inline server/plugin/api.go:1373]: why are we not checking error here?

File: server/plugin/api.go
Comment (ayusht2810) [inline server/plugin/api.go:1375]: approvalsCount or approversCount maybe?

File: server/plugin/api.go
Comment (ayusht2810) [inline server/plugin/api.go:1404]: shouldn't there be `postID` instead of `mattermostUserID`?

File: server/plugin/client.go
Comment (ayusht2810) [inline server/plugin/client.go:283]: shouldn't we return nil here if err is present as a good practice?

Comment (avas27JTG) [reply server/plugin/client.go:320]: that was handled but yes considering the practise we should do that

File: server/plugin/api.go
Comment (avas27JTG) [reply server/plugin/api.go:1465]: this is an util method inside this user-ID is used for DM, check that

File: server/plugin/api.go
Comment (avas27JTG) [reply server/plugin/api.go:1436]: it should be "approvals" because we are checking if no of required approvals reached or not which is independent of no of approvers available

File: server/plugin/api.go
Comment (ayusht2810) [inline server/plugin/api.go:1474]:
```suggestion
		p.handlePipelineApprovalRequestUpdateError(fmt.Sprintf("Error in updating the post: %s", postID), mattermostUserID, err)
```

**Review by ayusht2810 — APPROVED** [review]

File: server/constants/routes.go
Comment (manojmalik20) [inline server/constants/routes.go:36]: Why did you move this constant to the section of `Azure API paths`?

File: server/plugin/api.go
Comment (manojmalik20) [inline server/plugin/api.go:888]:
```suggestion
			approvers += approvalStep.AssignedApprover.DisplayName + "\n"
```

File: server/plugin/api.go
Comment (manojmalik20) [inline server/plugin/api.go:915]:
```suggestion
					Type:  model.POST_ACTION_TYPE_BUTTON,
```
Please fix this in the whole repo.

File: server/plugin/api.go
Comment (manojmalik20) [inline server/plugin/api.go:1437]: This function should not be in api.go.

File: server/plugin/api.go
Comment (manojmalik20) [inline server/plugin/api.go:1445]:
```suggestion
	numApprovalsReached := 0
```

File: server/plugin/api.go
Comment (manojmalik20) [inline server/plugin/api.go:1448]:
```suggestion
	for _, step := range approvalSteps {
```

File: server/serializers/subscriptions.go
Comment (manojmalik20) [inline server/serializers/subscriptions.go:195]:
```suggestion
type ApprovalStep struct {
```

File: server/constants/routes.go
Comment (avas27JTG) [reply server/constants/routes.go:36]: I think it was mistakenly moved there by Raghav during the conflicts resolution

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**
